webview-ui/src/components/AnalysisPage.tsx

```diff
@@ -30,6 +30,8 @@ const AnalysisPage: React.FC = () => {
   const [expandedViolations, setExpandedViolations] = useState<Set<string>>(new Set());
   const [isWaitingForSolution, setIsWaitingForSolution] = useState(false);
   const [serverRunning, setServerRunning] = useState(false);
+  const [serverStarting, setServerStarting] = useState(false);
```

Comment:
Keeping the flags in sync is tedious - how about deriving them from state directly as language client - check [`isRunning` helper.](https://github.com/microsoft/vscode-languageserver-node/blob/8e6bf18f22b2e83b60a27a51447b99acf98c4381/client/src/common/client.ts#L1236)